MVVM/Model/PressedKeys.cs

```diff
@@ -212,10 +224,54 @@ private void HandleControlModeChange()
 			StopAll();
 		}
 
+		private void HandleDriveModeChangeToggle()
+		{
+			if (LocalSettings.Singleton.Joystick.RoverDriveController != (int)RoverDriveControllerSelector.Controller.DirectDrive) return;
+
+			if (!LocalSettings.Singleton.Joystick.ToggleableKinematics) return;
+
+			if (ControlMode != ControlMode.Rover) return;
+
+			if (Input.IsActionJustPressed("crab_mode", true))
+				KinematicMode = KinematicMode.Crab;
+			else if (Input.IsActionJustPressed("spinner_mode", true))
+				KinematicMode = KinematicMode.Spinner;
+			else if (Input.IsActionJustPressed("ackermann_mode", true))
+				KinematicMode = KinematicMode.Ackermann;
+			else if (Input.IsActionJustPressed("ebrake_mode", true))
+				KinematicMode = KinematicMode.EBrake;
+			
+			_roverDriveControllerPreset.Mode = KinematicMode;
+
+		}
+
+		void HandleDriveModeChangeHold()
+		{
+			if (LocalSettings.Singleton.Joystick.RoverDriveController != (int)RoverDriveControllerSelector.Controller.DirectDrive) return;
+
+			if (LocalSettings.Singleton.Joystick.ToggleableKinematics) return;
+
+			if (ControlMode != ControlMode.Rover) return;
+
+			KinematicMode tempKM = KinematicMode.Ackermann;
```

Comment:
It sure work, *but* this is more elegant i think 🤓 
```cs
		public KinematicMode KinematicMode
		{
			get => _kinematicMode;
			private set
			{
				if(_kinematicMode == value) return; // <- over here
				_kinematicMode = value;
				EventLogger.LogMessage("PressedKeys", EventLogger.LogLevel.Info, $"Kinematic Mode changed {value}");
				OnKinematicModeChanged?.Invoke(value);
			}
		}
```

Reply:
It sure is. Adressed in [More elegance](https://github.com/wisniax/RoverControlApp/pull/36/commits/bf4a8afa2c09c0c126da07ab5d15632cfceddb8a)